Build and Run on arm platforms is broken

Hi Xianyi,

The build of OpenBLAS in latest develop branch  is completely broken on arm platforms.
I always get bus-errors, tested on three different arm platforms.
I don't know the reason, It could be the merge, that introduced common_stackalloc.h.

I think, that you should revert the develop branch to a stable version.

Best regards
Werner

It's possible to build and test an `arm64` on Debian Linux with:

    sudo dpkg --add-architecture arm64
    sudo apt-get update
    sudo apt-get install qemu-user-binfmt gfortran-5-aarch64-linux-gnu libgfortran3:arm64
    make TARGET=ARMV8  CC=aarch64-linux-gnu-gcc-5 FC=aarch64-linux-gnu-gfortran-5

Then apply the following patch to enable make tests:

```patch
--- a/Makefile
+++ b/Makefile
@@ -112,8 +112,6 @@ endif
 
 tests :
 ifndef NOFORTRAN
-ifndef TARGET
-ifndef CROSS
 	touch $(LIBNAME)
 ifndef NO_FBLAS
 	$(MAKE) -C test all
@@ -125,8 +123,6 @@ ifndef NO_CBLAS
 	$(MAKE) -C ctest all
 endif
 endif
-endif
-endif
```

`make tests` will be long because it's pure CPU emulation.

I don't have any failure in `make test` doing that.

I also tried with `arm`:

    sudo dpkg --add-architecture armhf
    sudo apt-get update
    sudo apt-get install qemu-user-binfmt gfortran-5-arm-linux-gnueabihf libgfortran3:armhf
    make TARGET=ARMV8 CC=arm-linux-gnueabihf-gcc-5 CC=arm-linux-gnueabihf-gfortran-5

but building fail with:

    make[1]: Entering directory '/home/jerome/OpenBLAS-arm/kernel'
    make[1]: *** No rule to make target '../kernel/arm/amax.S', needed by 'samax_k.o'.  Stop.

--- a/common_stackalloc.h
+++ b/common_stackalloc.h
@@ -29,8 +29,8 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #ifdef STACK_ALLOC_PROTECT
 // Try to detect stack smashing
 #include <assert.h>
-#define STACK_ALLOC_PROTECT_SET volatile BLASLONG stack_check = 0x7ff8010203040506;
-#define STACK_ALLOC_PROTECT_CHECK assert(stack_check == 0x7ff8010203040506);
+#define STACK_ALLOC_PROTECT_SET volatile int stack_check = 0x7fc01234;
+#define STACK_ALLOC_PROTECT_CHECK assert(stack_check == 0x7fc01234);
 #else
 #define STACK_ALLOC_PROTECT_SET
 #define STACK_ALLOC_PROTECT_CHECK

